### PR TITLE
fix(active-hint): re-implement active-hint

### LINKED ISSUE #493: Active hint moves away from active floating window when tiling actions are performed
**Distribution (run `cat /etc/os-release`):** Pop!_OS 20.04 LTS
 
**Related Application and/or Package Version (run `apt policy pop-shell` or provide the latest commit if building locally):** Occurs in `0.1.0~1597190322~20.04~149061a` (current master) as well as `0.1.0~1597190879~20.04~52ded06`.

**Issue/Bug Description:** After floating a window, the active hint usually moves back to a window that is not currently active. Additionally, if a window is closed and causes other tiled windows to be adjusted, and the focus is placed on a floating window, the active hint ends up going to one of the adjusted windows instead of the active floating window.

**Steps to reproduce (if you know):**
- Scenario 1:
  - Enable active hints.
  - Tile two windows.
  - Float one of the windows with Super-G.
- Scenario 2:
  - Arrange four windows in a tree, then float one of them.
  - Select the floated window, then one of the tiled windows.
  - Close the tiled window with Super-Q.

**Expected behavior:** The active hint should always surround the active window, even if the active window is floated.

**Other Notes:**
- Video demonstrating both scenarios: https://www.dropbox.com/s/eyjx08s6b9zhpcu/active-hint-issue.webm?dl=0

### PR DESCRIPTION
**Pull request by jmmaranan**
Simplified tracking of border for each window using St.Bin.

Closes #493. #487 looks ok

### CODE REVIEW

File: src/window.ts [hunk 1]
```diff
@@ -262,6 +272,46 @@ export class ShellWindow {
...
+                if (this.is_maximized() || this.meta.minimized) {
+                    this.hide_border();
+                }
+                this.update_border_layout();
+            }),
+            this.meta.connect('position-changed', () => {
+                this.update_border_layout();
+            }),
+        ];
+    }
```
Comment (mmstick) [inline src/window.ts:314]: Would this work here? https://github.com/pop-os/shell/blob/master_focal/src/extension.ts#L381:L386

File: src/extension.ts [hunk 2]
```diff
@@ -449,37 +443,37 @@ export class Ext extends Ecs.System<ExtEvent> {
...
+        //     let cws = this.workspace_id(null);
+
+        //     if (aws[0] === cws[0] && aws[1] === cws[1]) {
+        //         this.active_hint.track(active);
+        //     } else {
+        //         this.active_hint.untrack();
+        //     }
+        // };
+
+        // refocus_hint();
```
Comment (mmstick) [inline src/extension.ts:476]: We can remove all code that has been commented out.

File: src/window.ts (hunk 1 above)
Comment (jmmaranan) [reply src/window.ts:314]: It does not seem to work when I bounded the handler. That's why I added two new sets.

Comment (mmstick) [reply src/window.ts:314]: Then we should definitely add these signals somewhere so we can disconnect them on extension disabling / window destruction.

File: src/extension.ts (hunk 2 above)
Comment (jmmaranan) [reply src/extension.ts]: Removed all outdated code.

File: src/window.ts (hunk 1 above)
Comment (jmmaranan) [reply src/window.ts:314]: @mmstick - added on `this.ext.window_signals` push on location of binding

**Comment by jmmaranan** [comment]
@mmstick - please do not merge this yet. I found an issue when single window and move workspace down. Or single window open, move it to another workspace. The active hint was not moving.

Let me see if it was the previous `refocus_hint()` on active workspace changed.

**Comment by jacobgkau** [comment]
Expanding on part of my comment from #499, here is what a smart gapped window looks like on this branch:

Screenshot from 2020-08-18 15-58-13

And here is what a manually maximized (Super-M) window looks like:

Screenshot from 2020-08-18 15-58-18

I personally think this makes sense. The testing criteria includes the active hint being hidden for a maximized window, but I can imagine a user wanting the active hint to appear while they're using smart gaps-- with multiple displays, it might be unclear what's active if the hint is hidden. This behavior still allows them to hide the active hint for maximized windows manually.

I do see the issue with the active hint from a single window remaining when moving to a new workspace:

Screenshot from 2020-08-18 15-58-22

I'll wait for that to be addressed.

**Comment by jmmaranan** [comment]
@jacobgkau, @mmstick - this now can be reviewed. 

One more discovery that a St.ThemeNode can be implemented to add feat to change colors of the border or sizes. Will add that up in a future PR.

**Comment by jmmaranan** [comment]
@jacobgkau - added fix on need for border and window to be on same workspace when changing

**Comment by jmmaranan** [comment]
It is currently using the window clone border class. I will add or update the existing css.

**Comment by jmmaranan** [comment]
I am currently working on a way for users to customize it though. A limited color set and border size. We still have the stacking problem that's why thicker border becomes better when the window actor restacks at the top.

**Comment by mmstick** [comment]
Thanks! All commits have been rebased to master

**Pull request closed without merging**